src/utils/validate-object-id-hook.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const errors = require('feathers-errors');
+const mongoose = require('mongoose');
+
+var validateObjectId = function() {
+  return function(hook) {
+    const id = hook.id;
+    if (id && !mongoose.Types.ObjectId.isValid(id)) {
```

Comment:
Actually, it depends. If we would like to enforce that the `:id` in `GET /photos/:id` must be an ObjectID, it should return `400 BadRequest` with `Cast to ObjectId failed`. 

However, imo, a REST consumer should not be aware of the underlying technology (ObjectID in MongoDB) except we document the format of `:id` properly.

Alternative way is using an alternative id field as the primary query id instead of the default `_id`. I ever discussed with the feathers team once at feathersjs/feathers-mongoose#102 and feathersjs/feathers-mongoose#103.